### PR TITLE
Make possible to set NULL values using config:set command

### PR DESCRIPTION
**Pull request by jalogut**
### Description
Currently is not possible to set `null` values using `config:set` command. That is needed, if we want to overwrite `core_config_data` values using `config.php` and `env.php` files.

This PR makes that possible by replacing "NULL" param value with `null` before the value is saved into the config files.

### Manual testing scenarios

1. `bin/magento config:set <path> NULL`
2. I would expect this value set to `null`. However, the current status is that "NULL" string is saved instead.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by hostep** [comment]
What if you want to use the actual word "NULL", that won't be possible anymore? (I realize almost nobody will want to do this, but still ...)

In the n98/magerun2 project, they fixed it with an extra flag `--no-null`, which change the interpretation when you provide `NULL`, either the value `null` or the word "NULL", maybe that's an idea which you can use over here as well?
See: https://github.com/netz98/n98-magerun2/blob/fb75eb2/src/N98/Magento/Command/Config/Store/SetCommand.php#L77-L86

**Comment by jalogut** [comment]
Yes, I know about the flag on n98. It’s just that I do not think anybody will need “NULL” as a string. I also think that it should be avoided, so I did not want to provide this option. If you think that it is needed, I wouldn’t have any problem to add it.

**Comment by hostep** [comment]
I also believe almost nobody will need "NULL" as a string, but still, one day, someone will probably need it for some obscure reason.
Let's wait for someone from the Magento team to give some feedback here to see if the extra flag is needed or not.

**Review by orlangur — CHANGES_REQUESTED** [review]
I don't like that we should use capsed `NULL` here and that string behavior is changed.

Can we have a separate command like `bin/magento config:unset <path>` or `bin/magento config:set-null <path>` maybe?

**Comment by orlangur** [comment]
Closing due to inactivity. Feel free to reach me out anytime later if you wish to continue work on this pull request and it will be reopened.

**Comment by jalogut** [comment]
Hi @orlangur

I’d rather go for the option `—no-null` as @hostep mentioned. Adding a especific command just for null values doesn't seem like the right solution for me. What do you think? 

If you use config propagation, you need to assign `NULL` value to unset config settings. With config propagation files, you configure values but not different actions depending on the value.

**Comment by joshuaadickerson** [comment]
unset seems best. Makes sure nobody will mistake it.

**Comment by orlangur** [comment]
Raised in Slack.

**Comment by orlangur** [comment]
Hi @jalogut, among maintainers the preferred version is `bin/magento config:unset <path>`, as this is a new feature please create a PR against `2.3-develop`.

**Pull request closed without merging**
